reconcile/utils/smtp_client.py

```diff
@@ -3,24 +3,55 @@
 from email.mime.multipart import MIMEMultipart
 from email.mime.text import MIMEText
 from email.utils import formataddr
-from typing import Any, Iterable, Union
+from typing import Iterable, Optional
+from pydantic import BaseModel
 
 from sretoolbox.utils import retry
 
-from reconcile.utils.secret_reader import SecretReader
+from reconcile.utils.secret_reader import SecretReader, SupportsSecret
 
 
-class SmtpClient:
-    def __init__(self, settings: dict[str, Any]) -> None:
-        smtp_config = self.get_smtp_config(settings)
-        self.host: str = smtp_config["server"]
-        self.port: int = int(smtp_config["port"])
-        self.user: str = smtp_config["username"]
-        self.passwd: str = smtp_config["password"]
-        self.mail_address: str = settings["smtp"]["mailAddress"]
-        self.timeout: int = settings["smtp"].get("timeout", 30)
+class SmtpCredentials(BaseModel):
+    server: str
+    port: int
+    username: str
+    password: str
+    # require_tls: bool - currently not in use
+
 
-        self._client: Union[smtplib.SMTP, None] = None
+def get_smtp_credentials(
+    secret_reader: SecretReader, secret: SupportsSecret
+) -> SmtpCredentials:
+    """Retrieve SMTP credentials from config or vault."""
+    # This will change later when SecretReader fully supports 'SupportsSecret'
```

Comment:
At this time, the usage of `SupportsSecret` is rather obscure 😅 Can you elaborate somewhere how it is going to be used. Maybe in the protocol definition :)